src/microbe_stage/editor/CellEditorComponent.Callbacks.cs

```diff
@@ -294,8 +294,9 @@ private void DoOrganelleUpgradeAction(OrganelleUpgradeActionData data)
     {
         data.UpgradedOrganelle.Upgrades = data.NewUpgrades;
 
-        // Uncomment when upgrades can visually affect the cell
-        // UpdateAlreadyPlacedVisuals();
+        microbeVisualizationOrganellePositionsAreDirty = true;
+
+        OnOrganellesChanged();
 
         UpdateStats();
```

Comment:
Now that `OnOrganellesChanged` is called from here, I think the update stats and StartAutoEvoPrediction calls need to be removed as they are basically duplicates as organelles changing should already trigger those actions.